packages/grafana-prometheus/src/querycache/QueryCache.ts

```diff
@@ -207,6 +207,7 @@ export class QueryCache<T extends SupportedQueryTypes> {
             if (amendedTable) {
               for (let i = 0; i < amendedTable.length; i++) {
                 cachedFrame.fields[i].values = amendedTable[i];
+                cachedFrame.fields[i].config = respFrame.fields[i].config;
```

Comment:
main thing to verify with this change is that `structureRev` does not increment excessively. it should stay stable across data refreshes when the field structures and config options do not change.

you can stick a `console.log()` into GraphNG to make sure we're not re-initializing the chart more than necessary.

https://github.com/grafana/grafana/blob/125a8a41dd6af4b5da86094d893d3bac429a6e6b/public/app/core/components/GraphNG/GraphNG.tsx#L210

Reply:
I did what you've suggested and it's just normal. Like it was before, so I'd say no impact on reinitialization.

Reply:
I'm seeing a few issues here, first you should only update the displayNameFromDS. 
```
if(cachedFrame.fields[i].config.displayNameFromDS !== respFrame.fields[i].config.displayNameFromDS){
                  cachedFrame.fields[i].config.displayNameFromDS = respFrame.fields[i].config.displayNameFromDS;
                }
```

Initial query:
<img width="1728" alt="image" src="https://github.com/user-attachments/assets/a30c24aa-8fad-450e-ba6f-ba3168cd1999">

subsequent incremental queries:
<img width="1728" alt="image" src="https://github.com/user-attachments/assets/5613116c-cdfa-4039-82b2-3dc6ab679838">

Second, I'm seeing an issue with base incremental querying with rate queries.
I can't reproduce it in ops, but I can locally with the same datasource on this branch:
https://ops.grafana-ops.net/d/cdz3iu7slcvlsf/incremental-query-repro?from=now-12h&to=now&timezone=utc&refresh=&editPanel=1

Local:

Query:
```
sum(rate(adaptive_logs_bytes_dropped_total{job="loki-dev-005/distributor", container="distributor"}[$__rate_interval]))
```
DS: Dev-cortex

<img width="1308" alt="image" src="https://github.com/user-attachments/assets/4d4c970b-46c3-41b2-9504-5450c3e739e4">
<img width="1312" alt="image" src="https://github.com/user-attachments/assets/14b82746-4f20-475f-a378-268b72c97c80">

If we log out the field config, we can see that the cached field has an interval of 30s, and the respFrame has an interval of 10s, which is causing points to get unaligned, let me zoom in a bit on those screenshots:

<img width="81" alt="image" src="https://github.com/user-attachments/assets/e5950086-0278-4c16-853f-508e043cfff3">
<img width="128" alt="image" src="https://github.com/user-attachments/assets/2ba6b03f-9173-4505-9443-37bd4e42c002">

Ok that doesn't really help does it, see the video below instead:


https://github.com/user-attachments/assets/b9890b50-e52d-4487-8da9-c7138c7ab45b